index.js

```diff
@@ -7,14 +10,287 @@ function ask(questionText) {
+  ['strawberries','tomatoes'])
+let backProduce = new Location(
+  'back produce',
+  `
+  You're in the back produce -- looking at heaps of single avocadoes, and potatoes.
+  There's also big juicy tomatoes, and peppers. There are also bags of onions and
+  avocadoes as well.`,
+  ['front produce','right produce','left produce'],
+  ['avodaoes',' potatoes','big juicy tomatoes','peppers','onions','bag of onions','bag of avocadoes'])
+
+  // TODO add description
+  let leftProduce = new Location(
+  'left produce',
+  `
+  You're in the left produce -- tons of fruit!
+  The usual suspects are here including some funky ones.
+  Yes, hidden behind the passion fruit, are the mangoes and plums.
+  There are also a back of different berries! Take your pick!`,
+  ['checkout','back produce','front produce'],
+  ['passion fruit', 'mangoes', 'plums','blackberries', 'blueberries','strawberries'])
+
+  // TODO add description
+  let rightProduce = new Location(
+  'right produce',
+  `You're in the right produce -- we have bell peppers and corn here.`,
+  ['back produce','front produce'],
+  ['corn','green bell peppers','yellow bell peppers','red bell peppers'])
+  
+  // TODO add description
+  let checkout = new Location(
+  'checkout',
+  `So you're ready to check out. Go everything on your list?`,
+  ['left produce']
+  ['receipt'])
+
+// locations & location instance lookupTable
+// TODO implementation of "forgiving" location names that doesn't
+//      require adding all these possibilities into the Location 
+//      class instance's adjacent array.
+//      ISSUE: currently, adding locations here isn't enough, I need
+//      to update locations in each Location class instance as well.
+//      seems like a bad/poor pattern!
+const locationLookUp = {
+  "main entrance" : mainEntrace,
+  "cart room" : cartRoom,
+  //"cartroom" : cartRoom,
+  //"carts": cartRoom,
+  "front produce" : frontProduce,
+  //"front" : frontProduce,
+  "back produce" : backProduce,
+  //"back" : backProduce,
+  "left produce" : leftProduce,
+  //"left" : leftProduce,
+  //"left side" : leftProduce,
+  "right produce" : rightProduce,
+  //"right" : rightProduce,
+  //"right side" : rightProduce,
+  "checkout" : checkout
+}
+
+// TODO add items
+// items - these will simply be shopping list, cart (could extend to cash register)
+// TODO - what items will be untakable? 
+class Item {
+  constructor(name, description = '', contents = [], available = false, takeable = false) {
+    this.name = name
+    this.contents = contents
+    this.available = available
+    this.takeable = takeable
+  }
+}
+
+// Method to create the shopper's list 
+const SHOPPING_LIST_LENGTH = 5
+
+// create produce inventory
+let createProduceInventory = () => {
+  // merge inventories from all the produce locations
+  return frontProduce.inventory
+    .concat(backProduce.inventory)
+    .concat(leftProduce.inventory)
+    .concat(rightProduce.inventory)
```

Comment:
I found this to be a really cool way to create a room inventory!

Reply:
Thank you!